gyp/pylib/gyp/xcodeproj_file.py

```diff
@@ -155,6 +155,23 @@
   _new_sha1 = sha.new
 
 
+try:
+    basestring     # Python 2
+except NameError:  # Python 3
+    basestring = (str, )
```

Comment:
I was going to suggest to re-implement it as in the [six library](https://github.com/benjaminp/six/blob/master/six.py#L35-L53) but it requires a lot more changes to this file, wherever `basestring` is used. After a second thought, I think it's fine like this.